### DISCUSSION TITLE
pole figures in version 4.5

### DISCUSSION THREAD

**Discussion by LiZa111111111** [neperfepx/neper#554] [Q&A] [answered]
Hello,
I have just updated the 4.5.0 version to try the Pole Figure generation. 
The -V mode using "space pf" seems to run fine but it does not create the figures. See the attached file. Have I missed something?

![fig_err](https://user-images.githubusercontent.com/113169470/189350550-b964924a-9a8a-4a32-b18a-a10b9504b6d8.JPG)

**Comment by mkasemer** [comment, marked as answer]
What is printed in `img1.png`?

Further, you should consider installing the `asymptote` package (the error you are receiving is because it is not installed). For instance, on Ubuntu you can execute: `apt-get install asymptote`.

**Reply by LiZa111111111** [reply]
After installing the asymptote, it gave another issue, see below.
img1.png is not printed anywhere...

![fig_err_2](https://user-images.githubusercontent.com/113169470/189354691-da48c322-919b-44c1-be45-c321c084997f.JPG)

**Reply by mkasemer** [reply]
@LiZa111111111 - please copy terminal output directly, using the syntax for a code block (see: [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) ).

Neper states that it is writing a file `1.png`. What is in the folder when Neper is finished (i.e., what is listed when you execute the `ls` command)?

**Reply by LiZa111111111** [reply]
Thank you for your answer !
This is the thing, the error appears only when I activate "space pf". 
The image file is printed out without space pf function as it should. 
Using "space pf" no data is written, nothing in the folder when Neper is done.

**Reply by mkasemer** [reply]
I am unable to reproduce the error using the example provided in the tutorial:
```
neper -T -n 100 -morpho gg -oricrysym cubic -o n100
neper -V n100.tess -space pf -print img2
```
This yields expected output, including a reproduction of the first pole figure shown in the tutorial. Likewise, when I set the image size (attempting to reproduce your exact command):
```
neper -V n100.tess -space pf -imagesize 500:500 -print img3
```
I get the expected output: the same pole figure with reduced image dimensions.

What is the tessellation you are producing? Have you set a crystal symmetry (@rquey: indeed, is a crystal symmetry necessary to set during tessellation in order to produce a pole figure?).

**EDIT:** I just made the same tessellation without specifying crystal symmetry, and was able to make a pole figure. So no: it is not strictly necessary to do so (whether your plot is particularly meaningful or not is a different question).

**Reply by LiZa111111111** [reply]
Ok. Which Ubuntu version do you use? I have Ubuntu 20.04.5 LTS. Should I update it to 22.04 ?

**Reply by mkasemer** [reply]
I would be surprised if this is an Ubuntu issue (though I also use v20). I suspect other issues on your end, but @rquey will have to comment with thoughts.

**Reply by rquey** [reply]
Not an Ubuntu issue (I also am on 20.04).

@LiZa111111111, please try this:
```
neper -T -n 10
neper -V n10-id1.tess -space pf -imageformat asy -print img
asy -f png -render 4.428421 img.asy -o img.png
```

**Reply by LiZa111111111** [reply]
Thank you so much, it worked! I still do not understand why it was not working without adding this extra command. Have I missed to install any library connected to this issue?

**Reply by rquey** [reply]
@LiZa111111111, these commands are only for troubleshooting.

So, did you get the `img.png` fine?

Now, what if you run
```
neper -V n10-id1.tess -space pf -imageformat png -print img
```
which does the same, but calling `asy` internally.

**Reply by LiZa111111111** [reply]
I was little bit wrong, I meant that pdf-file format worked but still it did not printed the .png format. 
Since I got the message "failed to make .tfm", I found out that I had to install texlive-full to get the font-lib - it was missing on my system. 
After the installation, the .png file was printed as it should and I am able to generate the complete pole figure analyses using the description according to the Neper web-page. 
Thank you for your support!

**Reply by rquey** [reply]
Correct, thanks. I'll add this requirement to the documentation.

**Comment by majidkavousi** [comment]
Hi @rquey 
I have installed the latest version of Neper and I was trying to plot some pole figures from the neper Tutorials. At first I had the same issue that no images were generating. Then, I installed asymptote. But I'm getting the following error now (I have also installed texlive-full):
 neper -V n100.tess -space pf -print img2

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.5.1-4
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 16 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/majidkavousi/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -V loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] n100.tess -space pf -print img2
Info   : ---------------------------------------------------------------
Info   : Loading tessellation...
Info   :     [i] Parsing file `n100.tess'...
Info   :     [i] Parsed file `n100.tess'.
Info   : Printing pole figure...
Info   :   - Computing symbols...
Info   :   - Generating png file (1200x900 pixels)...
Info   :     [o] Writing file `img2.png'...
Wrong __data_start/_end pair
Aborted (core dumped)
Info   :     [o] Wrote file `img2.png'.
Info   : Printing scale...
Info   : Elapsed time: 0.161 secs.


Do you have any idea on this issue?
![image](https://user-images.githubusercontent.com/99205979/228933108-8814d85a-7d36-4672-9ae6-b973dfd6203d.png)
